mysqld_exporter.go

```diff
@@ -181,47 +243,15 @@ func main() {
 	log.Infoln("Starting mysqld_exporter", version.Info())
 	log.Infoln("Build context", version.BuildContext())
 
-	dsn := os.Getenv("DATA_SOURCE_NAME")
+	dsn = os.Getenv("DATA_SOURCE_NAME")
 	if len(dsn) == 0 {
 		var err error
 		if dsn, err = parseMycnf(*configMycnf); err != nil {
 			log.Fatal(err)
 		}
 	}
 
-	collect := collector.Collect{
-		SlowLogFilter:        *slowLogFilter,
-		Processlist:          *collectProcesslist,
-		TableSchema:          *collectTableSchema,
-		InnodbTablespaces:    *collectInnodbTablespaces,
-		InnodbMetrics:        *collectInnodbMetrics,
-		GlobalStatus:         *collectGlobalStatus,
-		GlobalVariables:      *collectGlobalVariables,
-		SlaveStatus:          *collectSlaveStatus,
-		AutoIncrementColumns: *collectAutoIncrementColumns,
-		BinlogSize:           *collectBinlogSize,
-		PerfTableIOWaits:     *collectPerfTableIOWaits,
-		PerfIndexIOWaits:     *collectPerfIndexIOWaits,
-		PerfTableLockWaits:   *collectPerfTableLockWaits,
-		PerfEventsStatements: *collectPerfEventsStatements,
-		PerfEventsWaits:      *collectPerfEventsWaits,
-		PerfFileEvents:       *collectPerfFileEvents,
-		PerfFileInstances:    *collectPerfFileInstances,
-		UserStat:             *collectUserStat,
-		ClientStat:           *collectClientStat,
-		TableStat:            *collectTableStat,
-		QueryResponseTime:    *collectQueryResponseTime,
-		EngineTokudbStatus:   *collectEngineTokudbStatus,
-		EngineInnodbStatus:   *collectEngineInnodbStatus,
-		Heartbeat:            *collectHeartbeat,
-		HeartbeatDatabase:    *collectHeartbeatDatabase,
-		HeartbeatTable:       *collectHeartbeatTable,
-	}
-
-	c := collector.New(dsn, collect)
-	prometheus.MustRegister(c)
```

Comment:
After this change, registering collector every http request, we now call `Describe()` every http request, and as in result we call Collect() twice every http request.